types/src/crypto/asymmetric_key.rs

```diff
@@ -384,27 +376,59 @@ impl AsymmetricType for PublicKey {
     }
 
     fn secp256k1_from_bytes<T: AsRef<[u8]>>(bytes: T) -> Result<Self, Error> {
-        let mut public_key = k256::PublicKey::from_bytes(bytes.as_ref()).ok_or_else(|| {
-            Error::AsymmetricKey(format!(
-                "failed to construct secp256k1 public key.  Expected {} bytes, got {} bytes.",
-                Self::SECP256K1_LENGTH,
-                bytes.as_ref().len()
-            ))
-        })?;
-        public_key.compress();
-        Ok(PublicKey::Secp256k1(public_key))
+        match bytes.as_ref().len() {
+            Self::SECP256K1_LENGTH => {
+                let bytes =
+                    <[u8; Self::SECP256K1_LENGTH]>::try_from(bytes.as_ref()).map_err(|err| {
+                        Error::AsymmetricKey(format!(
+                            "failed to construct compressed secp256k1 public key. \
+                             Expected {} bytes, got {} bytes. \
+                             Error: {}",
+                            Self::SECP256K1_LENGTH,
+                            bytes.as_ref().len(),
+                            err
+                        ))
+                    })?;
+                Ok(PublicKey::Secp256k1(bytes))
+            }
+            secp256k1::util::FULL_PUBLIC_KEY_SIZE => {
+                let bytes = <[u8; secp256k1::util::FULL_PUBLIC_KEY_SIZE]>::try_from(bytes.as_ref())
+                    .map_err(|err| {
+                        Error::AsymmetricKey(format!(
+                            "failed to construct full secp256k1 public key. \
+                             Expected {} bytes, got {} bytes. \
+                             Error: {}",
+                            secp256k1::util::FULL_PUBLIC_KEY_SIZE,
+                            bytes.as_ref().len(),
+                            err
+                        ))
+                    })?;
+                let public_key = secp256k1::PublicKey::parse(&bytes).map_err(|err| {
+                    Error::AsymmetricKey(format!("Error constructing public key: {:?}", err))
+                })?;
+                Ok(PublicKey::Secp256k1(public_key.serialize_compressed()))
+            }
+            unsupported_number_of_bytes => Err(Error::AsymmetricKey(format!(
+                "failed to construct secp256k1 public key. \
+                 Unsupported number of bytes: {}",
+                unsupported_number_of_bytes
+            ))),
+        }
     }
 }
 
+// TODO: Change this to TryFrom, since all zero Secp256k1 secrets are forbidden.
```

Comment:
I don't think we should leave this as a todo - I believe it should be handled as part of this PR.

Reply:
As discussed, we should handle this in a follow up PR.